app/access.py

```diff
@@ -16,3 +17,20 @@ def wrapper(self, *args, **kwargs):
             return method(self, *args, **kwargs)
         return wrapper
     return check_acls
+
+
+def auth_or_token(method):
+    @functools.wraps(method)
+    def wrapper(self, *args, **kwargs):
+        data = self.get_json()
+        if 'token' in data:
+            matching_tokens = list(Token.query.filter(Token.id ==
+                                                      data['token']))
```

Comment:
`.one()` instead of `list(...)` will give you the matching item or error if there's more than one; `.first()` will just give the first one. I think either would work here.

Reply:
Actually: if this you're querying by the primary key (`id` in this case) you can just do `Token.query.get(data['token'])` as a shorthand.